planetscale/workflows.go

```diff
@@ -0,0 +1,211 @@
+	UpdatedAt      time.Time `json:"updated_at"`
+	RowsCopied     int64     `json:"rows_copied"`
+	RowsTotal      int64     `json:"rows_total"`
+	RowsPercentage int       `json:"rows_percentage"`
+}
+
+type WorkflowVDiff struct {
+	PublicID           string                     `json:"id"`
+	State              string                     `json:"state"`
+	CreatedAt          time.Time                  `json:"created_at"`
+	UpdatedAt          time.Time                  `json:"updated_at"`
+	StartedAt          *time.Time                 `json:"started_at"`
+	CompletedAt        *time.Time                 `json:"completed_at"`
+	HasMismatch        bool                       `json:"has_mismatch"`
+	ProgressPercentage int                        `json:"progress_percentage"`
+	EtaSeconds         int64                      `json:"eta_seconds"`
+	TableReports       []WorkflowVDiffTableReport `json:"table_reports"`
+}
+
+type WorkflowVDiffTableReport struct {
+	PublicID                   string                 `json:"id"`
+	TableName                  string                 `json:"table_name"`
+	Shard                      string                 `json:"shard"`
+	MismatchedRowsCount        int64                  `json:"mismatched_rows_count"`
+	ExtraSourceRowsCount       int64                  `json:"extra_source_rows_count"`
+	ExtraTargetRowsCount       int64                  `json:"extra_target_rows_count"`
+	ExtraSourceRows            map[string]interface{} `json:"extra_source_rows"`
+	ExtraTargetRows            map[string]interface{} `json:"extra_target_rows"`
+	MismatchedRows             map[string]interface{} `json:"mismatched_rows"`
+	SampleExtraSourceRowsQuery string                 `json:"sample_extra_source_rows_query"`
+	SampleExtraTargetRowsQuery string                 `json:"sample_extra_target_rows_query"`
+	SampleMismatchedRowsQuery  string                 `json:"sample_mismatched_rows_query"`
+	CreatedAt                  time.Time              `json:"created_at"`
+	UpdatedAt                  time.Time              `json:"updated_at"`
+}
+
+type ListWorkflowsRequest struct {
+	Organization string `json:"-"`
+	Database     string `json:"-"`
+}
+
+// WorkflowsService is an interface for interacting with the keyspace endpoints of the PlanetScale API
+type WorkflowsService interface {
+	List(context.Context, *ListWorkflowsRequest) ([]*Workflow, error)
+}
+
+type workflowsService struct {
+	client *Client
+}
+
+var _ WorkflowsService = &workflowsService{}
+
+func NeWorkflowsService(client *Client) *workflowsService {
+	return &workflowsService{client}
+}
+
+func (ws *workflowsService) List(ctx context.Context, listReq *ListWorkflowsRequest) ([]*Workflow, error) {
+	req, err := ws.client.newRequest(http.MethodGet, workflowsAPIPath(listReq.Organization, listReq.Database), nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating http request")
+	}
+
+	workflows := workflowsResponse{}
+
+	if err := ws.client.do(ctx, req, workflows); err != nil {
```

Comment:
@iheanyi workflows.Workflows is empty here in the test 🤔 would you know why?

Reply:
Missing the pointer reference, should be `&workflowsResponse{}` :)